apis/Google.Cloud.Firestore.Data/Google.Cloud.Firestore.Data.Snippets/UserGuideSnippets.cs

```diff
@@ -0,0 +1,387 @@
+                Population = 3900000L
+            };
+            DocumentReference document = await collection.AddAsync(city);
+
+            // Sample: SetAsyncOverwrite
+            City newCity = new City
+            {
+                Name = "Los Angeles",
+                Country = "United States of America",
+                Population = 3900005L
+            };
+            await document.SetAsync(newCity);
+            // End sample
+        }
+
+        [Fact]
+        public async Task SetAsyncMergeAll()
+        {
+            string projectId = _fixture.ProjectId;
+            FirestoreDb db = FirestoreDb.Create(projectId);
+
+            CollectionReference collection = db.Collection("cities");
+            City city = new City
+            {
+                Name = "Los Angeles",
+                Country = "USA",
+                State = "CA",
+                IsCapital = false,
+                Population = 3900000L
+            };
+            DocumentReference document = await collection.AddAsync(city);
+
+            // Sample: SetAsyncMergeAll
+            await document.SetAsync(
+                new { Country = "United States of America", Population = 3900005L },
+                SetOptions.MergeAll);
+            // End sample
+        }
+
+        [Fact]
+        public async Task SetAsyncMergeSpecific()
+        {
+            string projectId = _fixture.ProjectId;
+            FirestoreDb db = FirestoreDb.Create(projectId);
+
+            CollectionReference collection = db.Collection("cities");
+            City city = new City
+            {
+                Name = "Los Angeles",
+                Country = "USA",
+                State = "CA",
+                IsCapital = false,
+                Population = 3900000L
+            };
+            DocumentReference document = await collection.AddAsync(city);
+
+            // Sample: SetAsyncMergeSpecific
+            City newCity = new City
+            {
+                Name = "Los Angeles",
+                Country = "United States of America",
+                Population = 3900005L
+            };
+            await document.SetAsync(newCity, SetOptions.MergeFields("Country", "Population"));
+            // End sample
+        }
+
+        [Fact]
+        public async Task DeleteAsync()
+        {
+            string projectId = _fixture.ProjectId;
+            FirestoreDb db = FirestoreDb.Create(projectId);
+
+            CollectionReference collection = db.Collection("cities");
+            City city = new City
+            {
+                Name = "Los Angeles",
+                Country = "USA",
+                State = "CA",
+                IsCapital = false,
+                Population = 3900000L
+            };
+            DocumentReference document = await collection.AddAsync(city);
+
+            // Sample: DeleteAsync
+            await document.DeleteAsync();
+            // Multiple calls succeed, unless you specify a precondition.
```

Comment:
Clarify what this means? Does the second call silently do nothing?

Reply:
Done.